sycl/include/CL/sycl/ONEAPI/intel_matrix/matrix-amx.hpp

```diff
@@ -0,0 +1,438 @@
+// is the multiply of (TILE*TILE*4).
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout>
+struct joint_matrix<
+    Group, T, NumRows, NumCols, Layout,
+    typename std::enable_if<!((NumRows <= tile_size) &&
+                              (NumCols * sizeof(T) / 4 <= tile_size) &&
+                              (Layout != matrix_layout::col_major))>::type> {
+public:
+  // trows: Num of tiles in row.
+  // If T=int8, NumRows==33, trows should be 3=(33+15)/16
+  static constexpr size_t trows = (NumRows + tile_size - 1) / tile_size;
+  // tcols: Num of tiles in column.
+  static constexpr size_t tcols =
+      (NumCols * sizeof(T) / 4 + tile_size - 1) / tile_size;
+  // if T=int8, NumRows==33, NumCols==33*4, tile_size==16, then size of
+  // raw_storage should be 48*48*4.
+  // FIXME: Greedy Regalloc for tile seems has some limitation and currently we
+  // do tileload for (16,16*4) instead of varying shapes, so raw_storage's size
+  // is multiple of (16*16*4)
+  static constexpr size_t size = trows * tcols * tile_size * tile_size * 4;
+  // stride is aligned to T instead of int8
+  static constexpr size_t stride = tcols * tile_size * 4 / sizeof(T);
+  int8_t raw_storage[size];
+  static constexpr bool isSmall = false;
+
+public:
+  matrix_layout layout;
+  // We do zero-padding for matrix whose size is not fitted into tiles in ctor.
+  joint_matrix(Group sg) { memset(raw_storage, 0x00, size); }
+};
+
+// This template specialization handles cases where matrix can be put into a
+// tile and users specify layout is packed_a or packed_b
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout>
+struct joint_matrix<
+    Group, T, NumRows, NumCols, Layout,
+    typename std::enable_if<(NumRows <= tile_size) &&
+                            (NumCols * sizeof(T) / 4 <= tile_size)>::type> {
+public:
+  static constexpr size_t trows = (NumRows + tile_size - 1) / tile_size;
+  // tcols: Num of tiles in column.
+  static constexpr size_t tcols =
+      (NumCols * sizeof(T) / 4 + tile_size - 1) / tile_size;
+  static constexpr size_t size = trows * tcols * tile_size * tile_size * 4;
+  // stride is aligned to T instead of int8
+  static constexpr size_t stride = tcols * tile_size * 4 / sizeof(T);
+  _tile1024i tile;
+  static constexpr bool isSmall = true;
+  matrix_layout layout;
+  // We do zero-padding for matrix whose size is not fitted into tiles in ctor.
+  joint_matrix(Group sg) {}
+};
+
+} // namespace matrix
+
+namespace detail {
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix::matrix_layout Layout>
+inline __SYCL_ALWAYS_INLINE static
+    typename std::enable_if<(NumRows > matrix::tile_size) ||
+                                (NumCols * sizeof(T) / 4 > matrix::tile_size),
+                            void>::type
+    submatrix_load(detail::submatrix<T> &sub_m,
```

Comment:
Just curious: why not `submatrix::load` instead?
And while it can be a static member, could it be a member function?

Reply:
Hi, @keryell. Thanks for your comments. I really agree we should change into member function but we are reaching a deadline of release. So, could we merge this patch and Address your comments in our next PR?

Reply:
Whatever is good. Just need the tersest SYCL syntax at the end. :-)